src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2144,7 +2163,10 @@ private void GenerateOverloadCall(Function function)
                 string.Join(", ",
                     function.Parameters.Where(
                         p => p.Kind == ParameterKind.Regular).Select(
-                            p => p.Ignore ? p.DefaultArgument.String : p.Name)));
+                        p => p.Ignore ? ((p.Type.IsPointerToPrimitiveType() && p.Usage == ParameterUsage.InOut && p.HasDefaultValue) ? "ref param" + j++ : p.DefaultArgument.String)
```

Comment:
This has got too long and complicated. Please extract it to a separate function.